aksel.nav.no/website/pages/eksempler/actionmenu/filter.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { ChevronDownIcon } from "@navikt/aksel-icons";
+import { ActionMenu, Button } from "@navikt/ds-react";
+import { withDsExample } from "@/web/examples/withDsExample";
+
+const Example = () => {
+  const [views, setViews] = React.useState({
+    started: true,
+    fnr: false,
+    tags: true,
+  });
+  const [rows, setRows] = React.useState<string>("5");
+
+  const handleCheckboxChange = (checkboxId: string) => {
+    setViews((prevState) => ({
+      ...prevState,
+      [checkboxId]: !prevState[checkboxId],
+    }));
+  };
+
+  return (
+    <ActionMenu defaultOpen>
+      <ActionMenu.Trigger>
+        <Button
+          variant="secondary-neutral"
+          icon={<ChevronDownIcon aria-hidden />}
+          iconPosition="right"
+        >
+          Filter
+        </Button>
+      </ActionMenu.Trigger>
+      <ActionMenu.Content>
+        <ActionMenu.Group label="Kolonner">
+          <ActionMenu.CheckboxItem
+            checked={
+              Object.values(views).every(Boolean)
+                ? true
+                : Object.values(views).some(Boolean)
+                  ? "indeterminate"
+                  : false
+            }
+            onCheckedChange={() =>
+              Object.values(views).every(Boolean)
+                ? setViews((prevState) =>
+                    Object.keys(prevState).reduce(
+                      (acc, key) => ({ ...acc, [key]: false }),
+                      {} as typeof views,
+                    ),
+                  )
+                : setViews((prevState) =>
+                    Object.keys(prevState).reduce(
+                      (acc, key) => ({ ...acc, [key]: true }),
+                      {} as typeof views,
+                    ),
+                  )
```

Comment:
If you put `Object.values(views).every(Boolean)` into a const, you could simplify this code:
```suggestion
              setViews((prevState) =>
                Object.keys(prevState).reduce(
                  (acc, key) => ({ ...acc, [key]: !allChecked }),
                  {} as typeof views,
                ),
              )
```

Reply:
Fixed 🧠